packages/fast-components-react-base/src/toolbar/toolbar.tsx

```diff
@@ -0,0 +1,274 @@
+        if (this.props.enableAutoFocus) {
+            this.focus();
+        }
+    }
+
+    /**
+     * Brings focus to the appropriate menu-item
+     */
+    public focus(): void {
+        this.setFocus(this.state.focusIndex === -1 ? 0 : this.state.focusIndex, 1);
+    }
+
+    /**
+     * Create class names
+     */
+    protected generateClassNames(): string {
+        const {
+            toolbar,
+            toolbar__horizontal,
+            toolbar__vertical,
+        }: ToolbarClassNameContract = this.props.managedClasses;
+        const isVertical: boolean = this.props.orientation === Orientation.vertical;
+
+        return super.generateClassNames(
+            classNames(
+                this.props.managedClasses.toolbar,
+                [toolbar__vertical, isVertical],
+                [toolbar__horizontal, !isVertical]
+            )
+        );
+    }
+
+    /**
+     * Render all child elements
+     */
+    private renderChildren(): React.ReactChild[] {
+        return React.Children.map(this.props.children, this.renderChild);
+    }
+
+    /**
+     * Render a single child
+     */
+    private renderChild = (
+        child: React.ReactElement,
+        index: number
+    ): React.ReactChild => {
+        return React.cloneElement(child, {
+            tabIndex: index === this.state.focusIndex ? 0 : -1,
+        });
+    };
+
+    /**
+     * Determines if a given element should be focusable by the menu
+     */
+    private isFocusableElement = (element: Element): element is HTMLElement => {
+        return (
+            element instanceof HTMLElement &&
+            this.props.focusableRoles.indexOf(element.getAttribute("role")) !== -1 &&
+            (this.props.allowFocusOnDisabledItems || !this.isDisabledElement(element))
+        );
+    };
+
+    private isDisabledElement = (element: Element): element is HTMLElement => {
+        return (
+            element instanceof HTMLElement &&
+            (element.getAttribute("aria-disabled") === "true" ||
+                (element.hasAttribute("disabled") &&
+                    element.getAttribute("disabled") !== "false"))
+        );
+    };
+
+    /**
+     * Return an array of all focusabled elements that are children
+     * of the context menu
+     */
+    private domChildren(): Element[] {
+        return canUseDOM() && this.rootElement.current instanceof HTMLElement
+            ? Array.from(this.rootElement.current.children)
+            : [];
+    }
+
+    /**
+     * Ensure we always validate our internal state on item focus events, otherwise
+     * the component can get out of sync from click events
+     */
+    private handleItemFocus = (e: React.FocusEvent<HTMLElement>): void => {
+        const target: Element = e.currentTarget;
+        const focusIndex: number = this.domChildren().indexOf(target);
+
+        if (focusIndex === -1) {
+            return;
+        }
+
+        if (this.isDisabledElement(target)) {
+            target.blur();
```

Comment:
Would this cause focus and blur to fire on the control? Is it possible styling or logic might happen in the child that would be different from a normal disabled case where it never got focus anyway? Want to make sure we don't see flashes like styling into focus state and then not really fast.

Reply:
not doing this in new version, pr soon